source/include/azure_iot_json_reader.h

```diff
@@ -0,0 +1,182 @@
+ * @retval other Initialization failed.
+ *
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_Init( AzureIoTJSONReader_t * pxReader,
+                                                   const uint8_t * pucBuffer,
+                                                   uint16_t usBufferLen );
+
+/**
+ * @brief De-initializes an #AzureIoTJSONReader_t
+ *
+ * @param[in] pxReader A pointer to an #AzureIoTJSONReader_t instance to de-initialize
+ *
+ * @return An #AzureIoTHubClientResult_t value indicating the result of the operation.
+ * @retval #eAzureIoTHubClientSuccess The #AzureIoTJSONReader_t is de-initialized successfully.
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_Deinit( AzureIoTJSONReader_t * pxReader );
+
+/**
+ * @brief Reads the next token in the JSON text and updates the reader state.
+ *
+ * @param[in] pxReader A pointer to an #AzureIoTJSONReader_t instance containing the JSON to
+ * read.
+ *
+ * @return An #AzureIoTHubClientResult_t value indicating the result of the operation.
+ * @retval #eAzureIoTHubClientSuccess The token was read successfully.
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_NextToken( AzureIoTJSONReader_t * pxReader );
+
+/**
+ * @brief Reads and skips over any nested JSON elements.
+ *
+ * @param[in] pxReader A pointer to an #AzureIoTJSONReader_t instance containing the JSON to
+ * read.
+ *
+ * @return An #AzureIoTHubClientResult_t value indicating the result of the operation.
+ * @retval #eAzureIoTHubClientSuccess The children of the current JSON token are skipped successfully.
+ *
+ * @remarks If the current token kind is a property name, the reader first moves to the property
+ * value. Then, if the token kind is start of an object or array, the reader moves to the matching
+ * end object or array. For all other token kinds, the reader doesn't move and returns #eAzureIoTHubClientSuccess.
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_SkipChildren( AzureIoTJSONReader_t * pxReader );
+
+/**
+ * @brief Gets the JSON token's boolean value.
+ *
+ * @param[in] pxReader A pointer to an #AzureIoTJSONReader_t instance.
+ * @param[out] pxValue A pointer to a boolean to receive the value.
+ *
+ * @return An #AzureIoTHubClientResult_t value indicating the result of the operation.
+ * @retval #eAzureIoTHubClientSuccess The boolean value is returned.
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_TokenBoolGet( AzureIoTJSONReader_t * pxReader,
+                                                           bool * pxValue );
+
+/**
+ * @brief Gets the JSON token's number as a 32-bit signed integer.
+ *
+ * @param[in] pxReader A pointer to an #AzureIoTJSONReader_t instance.
+ * @param[out] pilValue A pointer to a variable to receive the value.
+ *
+ * @return An #AzureIoTHubClientResult_t value indicating the result of the operation.
+ * @retval #eAzureIoTHubClientSuccess The number is returned.
+ */
+AzureIoTHubClientResult_t AzureIoTJSONReader_TokenInt32Get( AzureIoTJSONReader_t * pxReader,
```

Comment:
Convention followed in all other place is <FileName>_<Verb><>()

her we are ending it with verb. Do we want that?

Reply:
good catch